protokube/cmd/protokube/main.go

```diff
@@ -172,7 +178,16 @@ func run() error {
 	var dnsScope dns.Scope
 	var dnsController *dns.DNSController
 	{
-		dnsProvider, err := dnsprovider.GetDnsProvider(dnsProviderId, nil)
+		var file io.Reader
+		if dnsProviderId == k8scoredns.ProviderName {
+			var lines []string
+			lines = append(lines, "etcd-endpoints = "+dnsServer)
+			lines = append(lines, "zones = "+zones[0])
+			config := "[global]\n" + strings.Join(lines, "\n") + "\n"
+			file = bytes.NewReader([]byte(config))
```

Comment:
AWS or GCE's DNS server doesn't need any specific information because they have public server address to access. But to use CoreDNS, we need to pass information about the IP address of the CoreDNS's etcd and zones to query. These are required to start a CoreDNS client. And this config file (it's not a real file though) is used to pass in these information. Without this config file, the DNS provider won't know which CoreDNS server to talk to.